File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/internal/screens/helper/FragmentSwitcher.java
package ru.usedesk.knowledgebase_gui.internal.screens.helper;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import java.util.List;

public class FragmentSwitcher {

    public static void switchFragment(@Nullable Fragment parentFragment,
                                      @NonNull Fragment fragment,
                                      int idContainer) {
        if (parentFragment != null) {
            parentFragment.getChildFragmentManager()
                    .beginTransaction()
                    .addToBackStack("cur")
                    .replace(idContainer, fragment)
                    .commit();
        }
    }

    public static boolean onBackPressed(@Nullable Fragment parentFragment) {
        if (parentFragment != null) {
            int count = parentFragment.getChildFragmentManager()
                    .getBackStackEntryCount();
            if (count > 1) {
                parentFragment.getChildFragmentManager()
                        .popBackStack();
                return true;
            }
        }
        return false;
    }

    @Nullable
    public static Fragment getLastFragment(@NonNull Fragment fragment) {
        List<Fragment> fragments = fragment.getChildFragmentManager()
                .getFragments();
        if (fragments.size() > 0) {
            return fragments.get(fragments.size() - 1);
        }

        return null;
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/helper/FragmentSwitcher.kt
package ru.usedesk.knowledgebase_gui.helper

import androidx.fragment.app.Fragment

object FragmentSwitcher {
    fun switchFragment(parentFragment: Fragment?,
                       fragment: Fragment,
                       idContainer: Int) {
        parentFragment?.childFragmentManager?.beginTransaction()
                ?.addToBackStack("cur")
                ?.replace(idContainer, fragment)
                ?.commit()
    }

    fun onBackPressed(parentFragment: Fragment?): Boolean {
        if (parentFragment != null) {
            val count = parentFragment.childFragmentManager.backStackEntryCount
            if (count > 1) {
                parentFragment.childFragmentManager.popBackStack()
                return true
            }
        }
        return false
    }

    fun getLastFragment(fragment: Fragment): Fragment? {
        val fragments = fragment.childFragmentManager.fragments
        return if (fragments.isNotEmpty()) {
            fragments.last()
        } else null
    }
}